vpr/src/place/annealer.cpp

```diff
@@ -0,0 +1,898 @@
+            delta_c = analyze_setup_slack_cost(setup_slacks_, placer_state_) * costs_.timing_cost_norm;
+        } else if (place_algorithm == e_place_algorithm::CRITICALITY_TIMING_PLACE) {
+            /* Take delta_c as a combination of timing and wiring cost. In
+             * addition to `timing_tradeoff`, we normalize the cost values */
+            VTR_LOGV_DEBUG(g_vpr_ctx.placement().f_placer_debug,
+                           "\t\tMove bb_delta_c %e, bb_cost_norm %e, timing_tradeoff %f, "
+                           "timing_delta_c %e, timing_cost_norm %e\n",
+                           bb_delta_c,
+                           costs_.bb_cost_norm,
+                           timing_tradeoff,
+                           timing_delta_c,
+                           costs_.timing_cost_norm);
+            delta_c = (1 - timing_tradeoff) * bb_delta_c * costs_.bb_cost_norm
+                      + timing_tradeoff * timing_delta_c * costs_.timing_cost_norm;
+        } else {
+            VTR_ASSERT_SAFE(place_algorithm == e_place_algorithm::BOUNDING_BOX_PLACE);
+            VTR_LOGV_DEBUG(g_vpr_ctx.placement().f_placer_debug,
+                           "\t\tMove bb_delta_c %e, bb_cost_norm %e\n",
+                           bb_delta_c,
+                           costs_.bb_cost_norm);
+            delta_c = bb_delta_c * costs_.bb_cost_norm;
+        }
+
+        NocCostTerms noc_delta_c; // change in NoC cost
+        /* Update the NoC data structure and costs*/
+        if (noc_opts_.noc) {
+            VTR_ASSERT_SAFE(noc_cost_handler_.has_value());
+            noc_cost_handler_->find_affected_noc_routers_and_update_noc_costs(blocks_affected_, noc_delta_c);
+
+            // Include the NoC delta costs in the total cost change for this swap
+            delta_c += calculate_noc_cost(noc_delta_c, costs_.noc_cost_norm_factors, noc_opts_);
+        }
+
+        // 1 -> move accepted, 0 -> rejected.
+        move_outcome = assess_swap_(delta_c, annealing_state_.t);
+
+        //Updates the manual_move_state members and displays costs to the user to decide whether to ACCEPT/REJECT manual move.
+#ifndef NO_GRAPHICS
+        if (manual_move_enabled) {
+            move_outcome = pl_do_manual_move(delta_c, timing_delta_c, bb_delta_c, move_outcome);
+        }
+#endif //NO_GRAPHICS
+
+        if (move_outcome == e_move_result::ACCEPTED) {
+            costs_.cost += delta_c;
+            costs_.bb_cost += bb_delta_c;
+
+            if (place_algorithm == e_place_algorithm::SLACK_TIMING_PLACE) {
+                // Update the timing driven cost as usual
+                costs_.timing_cost += timing_delta_c;
+
+                // Commit the setup slack information
+                // The timing delay and cost values should be committed already
+                commit_setup_slacks(setup_slacks_, placer_state_);
+            }
+
+            if (place_algorithm == e_place_algorithm::CRITICALITY_TIMING_PLACE) {
```

Comment:
Suggest putting this algorithm first, as it is the default.  Also can make an if ... else if instead of two ifs